tvb/simulator/models/epileptorcodim3.py

```diff
@@ -0,0 +1,595 @@
+        mu2 = self.R * (self.E[0] * numpy.cos(z) + self.F[0] * numpy.sin(z))
+        mu1 = -self.R * (self.E[1] * numpy.cos(z) + self.F[1] * numpy.sin(z))
+        nu = self.R * (self.E[2] * numpy.cos(z) + self.F[2] * numpy.sin(z))
+
+        # Computes x_s, which is the solution to x_s^3 - mu2*x_s - mu1 = 0
+        if self.N == 1:
+            xs = (mu1 / 2.0 + numpy.sqrt(mu1 ** 2 / 4.0 - mu2 ** 3 / 27.0 + 0 * 1j)) ** (
+                1.0 / 3.0) + (mu1 / 2.0 - numpy.sqrt(mu1 ** 2 / 4.0 - mu2 ** 3 / 27.0 + 0 * 1j)) ** (
+                1.0 / 3.0)
+        elif self.N == 2:
+            xs = -1.0 / 2.0 * (1.0 - 1j * 3 ** (1.0 / 2.0)) * (mu1 / 2.0 + numpy.sqrt(
+                mu1 ** 2 / 4.0 - mu2 ** 3 / 27.0 + 0 * 1j)) ** (1.0 / 3.0) - 1.0 / 2.0 * (
+                1.0 + 1j * 3 ** (1.0 / 2.0)) * (mu1 / 2.0 - numpy.sqrt(
+                mu1 ** 2 / 4.0 - mu2 ** 3 / 27.0 + 0 * 1j)) ** (1.0 / 3.0)
+        elif self.N == 3:
+            xs = -1.0 / 2.0 * (1.0 + 1j * 3 ** (1.0 / 2.0)) * (mu1 / 2.0 + numpy.sqrt(
+                mu1 ** 2 / 4.0 - mu2 ** 3 / 27.0 + 0 * 1j)) ** (1.0 / 3.0) - 1.0 / 2.0 * (
+                1.0 - 1j * 3 ** (1.0 / 2.0)) * (mu1 / 2.0 - numpy.sqrt(
+                mu1 ** 2 / 4.0 - mu2 ** 3 / 27.0 + 0 * 1j)) ** (1.0 / 3.0)
+        xs = numpy.real(xs)
+
+        xdot = -y
+        ydot = x ** 3 - mu2 * x - mu1 - y * (nu + self.b * x + x ** 2)
+        if self.modification:
+            zdot = -self.c * (
+            numpy.sqrt((x - xs) ** 2 + y ** 2) - self.dstar + 0.1 * (z - 0.5) ** 7 + self.Ks * coupling[0, :])
+        else:
+            zdot = -self.c * (numpy.sqrt((x - xs) ** 2 + y ** 2) - self.dstar + self.Ks * coupling[0, :])
+
+        derivative = numpy.array([xdot, ydot, zdot])
+        return derivative
+
+    def update_derived_parameters(self):
+        r"""
+        The equations were taken from [Saggioetal_2017]
+        cf. Eqn. (7), page 17
+
+        Here we parametrize the great arc which lies on a sphere of radius R between the points A and B, which are given by:
+            .. math::
+                A &= \begin{pmatrix}\mu_{2,start} & -\mu_{1,start} & \nu_{start} \end{pmatrix} \\
+                B &= \begin{pmatrix}\mu_{2,stop} & -\mu_{1,stop} & \nu_{stop} \end{pmatrix}
+
+        Then we parametrize this great arc with z as parameter by :math:`R(E \cos z + F \sin z)`
+            where the unit vectors E and F are given by:
+            .. math::
+                E &= A/\|A\| \\
+                F &= ((A \times B) \times A)/\|(A \times B) \times A\|
+        """
+
+        A = numpy.array([self.mu2_start[0], -self.mu1_start[0], self.nu_start[0]])
+        B = numpy.array([self.mu2_stop[0], -self.mu1_stop[0], self.nu_stop[0]])
+
+        self.E = A / numpy.linalg.norm(A)
+        self.F = numpy.cross(numpy.cross(A, B), A)
+        self.F = self.F / numpy.linalg.norm(self.F)
+
+
+class EpileptorCodim3_slowmod(Model):
```

Comment:
Is this a completely different model?

Reply:
This model is similar, it is a different implementation. It allows for multiple classes of bursters to be visited by a simulation. The model dynamics are essentially the same. However there are some differences in the input of the system and some extra dynamics to allow the transition of classes